Adding organization section (#29796)
keycloak/keycloak: docs/documentation/server_admin/topics/organizations/managing-organization.adoc

[id="managing-organization_{context}"]

> =  Managing an Organization

Reviewer: ```suggestion
=  Managing an organization
```

[role="_abstract"]

> Organizations can be managed once you enable the `Organizations` setting in the `Realm Settings` page.
> 
> Once you enable this setting, the `Organizations` section is available from the left side menu.
> 
> By clicking on the `Organizations` section you can start creating an organization by clicking on the `Create Organization`
> button. If organizations already exist, you will see a list with all of them and options to search, edit, or delete an organization.

Reviewer: ```suggestion

You enable the Organizations feature for the current realm.

.Procedure

. Click `Realm Settings` in the menu.

. Click *Create Organization*.
+
If organizations already exist, you see a list of organizations and options to search, edit, or delete an organization.
```

Reviewer: Pedro, for accessibility reasons, we avoid using  directional words such as left or right, because it makes no sense to a visually impaired user.

Author: Makes sense, will keep that in mind.

An organization has the following settings:

Name::
A user-friendly name for the organization.

Domains::
A set of one or more domains that belongs to this organization. The domain is mainly used to map users to an organization
based on their email domains as well as unlock different behaviors when managing and authenticating the organization members.

Description::
A free-text field to describe the organization.

> Once you create an organization, you are allowed to manage additional settings we will be looking at in the following sections:

Reviewer: ```suggestion
Once you create an organization, you can manage the additional settings that are described in the following sections:
```

> * Manage Attributes
> * Manage Members
> * Manage IdentityProviders
> 
> ==  Disabling an Organization

Reviewer: ```suggestion
* Manage attributes
* Manage members
* Manage identity providers

==  Disabling an organization
```

> You can disable an organization by clicking on the `Enabled` switch at the top of the page.

Reviewer: ```suggestion
To disable an organization toggle *Enabled*.
```

Reviewer: Is there another way to indicate where *Enabled* is, such as next to a certain field?

Author: Yes, I can improve that.

> When an organization is disabled, organization members are no longer permitted to authenticate to the realm.

Reviewer: ```suggestion
When an organization is disabled, organization members can no longer authenticate to the realm.
```

> == Remove an Organization
> 
> Remove an Organization

Reviewer: ```suggestion
== Remove an organization

Remove an organization
```

[id="managing-organization_{context}"]

[[_enabling_organization_]]
=  Enabling Organizations

> To use organizations, you need to enable the feature for the current realm.

Reviewer: I generally omit "need to" since it adds no meaning.

.Procedure

. Click *Realm Settings* in the menu.

. Toggle *Organizations* to *On*.

Once the feature is enabled, you are able to manage organizations through the `Organizations` section available from the
menu.

=  Managing an organization
[role="_abstract"]

> From the `Organizations` section you can manage all the organizations in your realm.

Reviewer: ```suggestion
From the *Organizations* section, you can manage all the organizations in your realm.
```

Author: Fixed by https://github.com/keycloak/keycloak/pull/29796/commits/6c7e048eac12627d5abe15abcfaaa62a5014573a.

.Procedure

. Click *Create Organization*.

If organizations already exist, you see a list of organizations and options to search, edit, or delete an organization.

An organization has the following settings:

Name::
A user-friendly name for the organization. The name is unique within a realm.

Domains::
A set of one or more domains that belongs to this organization. A domain cannot be shared by different organizations
within a realm.

Description::
A free-text field to describe the organization.

Once you create an organization, you can manage the additional settings that are described in the following sections:

* <<_managing_attributes_,Manage attributes>>
* <<_managing_members_,Manage members>>
* <<_managing_identity_provider_,Manage identity providers>>

== Understanding organization domains

When managing an organization, the domain associated with an organization plays an important role in how
organization members authenticate to a realm and how their profiles are validated.

> One of the key roles of a domain is to help to identify to which organization a user belongs to. By looking at their
> email address, {project_name} will match a corresponding organization using the same domain and eventually change the
> authentication flow based on the organization requirements.
> 

Reviewer: ```suggestion
One of the key roles of a domain is to help to identify to the organizations where a user is a member. By looking at their
email address, {project_name} will match a corresponding organization using the same domain and eventually change the
authentication flow based on the organization requirements.

```

The domain also allows organizations to enforce that users are not allowed to use a domain in their emails

> other than those associated with an organization. This is specially useful when users, and their identities, are federated from
> identity providers associated with an organization and you want to force a specific email domain for their email addresses.

Reviewer: ```suggestion
other than those associated with an organization. This restriction is especially useful when users, and their identities, are federated from
identity providers associated with an organization and you want to force a specific email domain for their email addresses.
```

==  Disabling an organization

To disable an organization, toggle *Enabled* to *Off*.

> When an organization is disabled, you are still able to manage it through the management interfaces, but their members
> will not be able to authenticate to the realm, including authenticating through the identity providers associated with the
> organization as they are also automatically disabled.

Reviewer: ```suggestion
When an organization is disabled, you can still manage it through the management interfaces, but the organization members
cannot authenticate to the realm, including authenticating through the identity providers associated with the
organization as they are also automatically disabled.
```

However, the unmanaged members of an organization are still able to authenticate to the realm as they are also realm users, but
tokens will not hold metadata about their relationship with an organization that is disabled.

> For more details about managed and unmanaged users, look at the <<_managed_unmanaged_members_,Managed and unmanaged members>> section.

Reviewer: ```suggestion
For more details about managed and unmanaged users, see <<_managed_unmanaged_members_,Managed and unmanaged members>> section.
```

== Remove an organization

To remove an organization, you can click on the `Delete` action for the corresponding organization in the listing page or

> when editing an organization.

Author: Fixed by https://github.com/keycloak/keycloak/pull/29796/commits/c2305e209a21822318a4f2fd861d42baa8684d88.

> When removing an organization, all data associated with it will be deleted, including any managed member. Be sure when performing
> this operation as any account originated from the identity providers associated with the organization will be deleted.

Reviewer: ```suggestion
When removing an organization, all associated data is deleted, including any managed member. Also, any account originated from the identity providers associated with the organization will be deleted.
```

Author: Fixed by https://github.com/keycloak/keycloak/pull/29796/commits/b6d268e022bee42c38148854e49e62ba999a2ce4.

> Unmanaged users and identity providers are going to be kept in the realm, but they will no longer be linked to the

Reviewer: ```suggestion
Unmanaged users and identity providers remain in the realm, but they are no longer linked to the
```

organization.

> For more details about managed and unmanaged users, look at the <<_managed_unmanaged_members_,Managed and unmanaged members>> section.

Reviewer: ```suggestion
For more details about managed and unmanaged users, see <<_managed_unmanaged_members_,Managed and unmanaged members>>.
```

Author: Fixed by https://github.com/keycloak/keycloak/pull/29796/commits/b6d268e022bee42c38148854e49e62ba999a2ce4.
